crate/server/src/core/kms.rs

```diff
@@ -119,10 +121,8 @@ impl KMS {
             };
 
             // Generate a new key
-            let key: Key<32> = {
-                let mut rng = self.rng.lock().expect("failed locking the RNG");
-                Key::<32>::new(&mut *rng)
-            };
+            let mut rng = CsRng::from_entropy();
```

Comment:
If there is a `rng` in `self` you should use it instead as instantiating a new RNG is really expensive.

Reply:
good point